### PR TITLE
Added second assignment

### PR DESCRIPTION
**Pull request by galeal2**
1) Find all ancestors of a key in a Binary Tree.
2) Find the lowest common ancestors of 2 nodes.

### CODE REVIEW

File: assignment_2_Q2.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+class Node:
```
Comment (ariava) [inline assignment_2_Q2.py]: Can you try to group the common classes in a separate library, or use assignment_2_Q1.py as a library for assignment_1_Q2.py, as you prefer? In this way you ensure that your code is what is called a "single point of truth". For example, let's say you find a bug in the BST class while you are coding the Q2 file. You must remember afterwards to replicate the fix in the Q1 file. If you avoid duplication of the code it's easier, there is only one point to change.

File: assignment_2_Q1.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            # we need to modify the parent nodes
+            if item < parent.item:
+                parent.left = Node(item, None, None)
+            else:
+                parent.right = Node(item, None, None)
+
+
+
+# find all ancestors for a Binary Tree
+def rec_find_ancestors(ptr, key, path):
```
Comment (ariava) [inline assignment_2_Q1.py]: What does "rec" stand for? Could you please use a more obvious term for the slow readers like me, or explain in a comment?

Also, can you write a short docstring that explains what the parameters are for?

File: assignment_2_Q1.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+class Node:
+    def __init__(self, item, left = None, right = None):
+        self.item = item
```
Comment (ariava) [inline assignment_2_Q1.py]: This is just a nit: wouldn't it be clearer to call it "key"?

File: assignment_2_Q1.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    if BT.root.item == key:
+        return None
+    if not rec_find_ancestors(ptr, key, []):
+        return None
+    return rec_find_ancestors(ptr, key, [])
+
+# tests
+def main():
+    
+    tree = BST([16, 9, 18, 3, 14, 19, 1, 5])
```
Comment (ariava) [inline assignment_2_Q1.py]: Great job with testing!

**Review by ariava — CHANGES_REQUESTED** [review]
Great job with the testing!

**Comment by galeal2** [comment]
I changed some variables' names, added docstring (i am not sure if that is right, I have never used it before) and I created different files for the Node class and BST class.
Thank you

**Review by ariava — APPROVED** [review]
Looks great! Well done!

**Pull request merged**